client/components/common/CardDocumentView.jsx

```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Row, Input } from 'react-materialize';
+import * as DocumentAction from '../../actions/documentAction';
+import { deleteFlashMessage } from '../../actions/flashMessages';
+
+class CardDocumentView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleView = this.handleView.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleDocumentOnchange = this.handleDocumentOnchange.bind(this);
+    this.handleEditDocumentSubmit = this.handleEditDocumentSubmit.bind(this);
+
+    this.state = {
+      title: this.props.document.title,
+      body: this.props.document.body,
+      access: this.props.document.access,
+      isLoading: false,
+    };
+  }
+  componentDidMount() {
+    $('.modal').modal();
+    $('.button-collapse').sideNav();
+    $('ul.tabs').tabs();
+    $('.dropdown-button').dropdown({
+      inDuration: 300,
+      outDuration: 225,
+      constrainWidth: false,
+      hover: false, // Activate on hover
+      gutter: 0, // Spacing from edge
+      belowOrigin: false, // Displays dropdown below the button
+      alignment: 'left',
+      stopPropagation: false, // Stops event propagation
+    },
+  );
+  }
+
+  handleView(event) {
+    event.preventDefault();
+    const docID = this.props.document.id;
+    const modal = `#viewDocument${docID}`;
+    $(modal).modal('open');
+  }
+
+  handleEdit(event) {
+    event.preventDefault();
+    const docID = this.props.document.id;
+    const modal = `#editDocument${docID}`;
+    $(modal).modal('open');
+  }
+
+  handleDelete() {
+    event.preventDefault();
+    const confirmDelete =
+      confirm('Are you sure you want to delete this Document');
+    if (confirmDelete === true) {
+      const documentId = this.props.document.id;
+      this.props.deleteDocument(documentId);
+      this.props.deleteFlashMessage(1);
+    }
+  }
+
+  handleDocumentOnchange(event) {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  handleEditDocumentSubmit(event) {
+    event.preventDefault();
+    const document = this.state;
+    const documentId = this.props.document.id;
+    this.props.updateDocument(document, documentId,
+      this.props.currentUser.roleId);
+    this.props.deleteFlashMessage(1);
+  }
+
+  render() {
+    const { document, myDocument, readOnly } = this.props;
+    const currentDate = moment(new Date());
+    const DateCreated = moment(document.createdAt);
+    const dateUpdated = moment(document.updatedAt);
+    const documentDate = DateCreated.from(currentDate);
+    const updatedDocumentDate = dateUpdated.from(currentDate);
+    const { title, body } = this.state;
+    return (
+    <div>
```

Comment:
I think it's a good idea to extract this div out to a presentation component, this will make the logic easy to follow.